skrub/_gap_encoder.py

```diff
@@ -645,6 +646,36 @@ class GapEncoder(TransformerMixin, BaseEstimator):
     fitted_models_: list[GapEncoderColumn]
     column_names_: list[str]
 
+    @classmethod
+    def _merge(cls, transformers_list: list[GapEncoder]):
+        # merge GapEncoder fitted on different columns
+        # into a single GapEncoder
+        # useful for parallelization in the TableVectorizer
+        full_transformer = clone(transformers_list[0])
+        # assert rho_ is the same for all transformers
+        rho_ = transformers_list[0].rho_
+        assert all([transformers.rho_ == rho_ for transformers in transformers_list])
```

Comment:
We prefer writing a clear error message instead of using `assert`. In which scenarios could we have `transformers.rho_ != rho_`?

If such scenarios don't exist, you can safely remove this assertion.

Reply:
Yes I don't have any scenario in mind, it was mostly to double check. I'm removing it.